lib/statsd/instrument/environment.rb

```diff
@@ -78,12 +78,12 @@ def statsd_default_tags
         env.key?("STATSD_DEFAULT_TAGS") ? env.fetch("STATSD_DEFAULT_TAGS").split(",") : nil
       end
 
-      def statsd_flush_interval
-        Float(env.fetch("STATSD_FLUSH_INTERVAL", StatsD::Instrument::BatchedUDPSink::DEFAULT_FLUSH_INTERVAL))
+      def statsd_buffer_capacity
+        Integer(env.fetch("STATSD_BUFFER_CAPACITY", StatsD::Instrument::BatchedUDPSink::DEFAULT_BUFFER_CAPACITY))
       end
 
-      def statsd_buffer_capacity
-        Float(env.fetch("STATSD_BUFFER_CAPACITY", StatsD::Instrument::BatchedUDPSink::DEFAULT_BUFFER_CAPACITY))
+      def statsd_batching?
+        statsd_buffer_capacity > 0 && Float(env.fetch("STATSD_FLUSH_INTERVAL", 1.0)) > 0.0
```

Comment:
`STATSD_FLUSH_INTERVAL` is deprecated but there's no deprecation message, maybe we could `warn` when we find a value here?